pkg/server/server.go

```diff
@@ -251,7 +252,28 @@ func (s *Server) Close() error {
 	return errors.Collect(ErrCloseServer, errs...)
 }
 
-func buildEtcd(cfg *config.Config, logger *zap.Logger, pubAddr string, engine *gin.Engine) (srv *embed.Etcd, err error) {
+func buildEtcd(cfg *config.Config, logger *zap.Logger, engine *gin.Engine) (srv *embed.Etcd, err error) {
+	if cfg.Cluster.ClusterName == "" {
+		return nil, errors.New("cluster_name can not be empty")
+	}
+	if cfg.Cluster.NodeName == "" {
+		cfg.Cluster.NodeName = fmt.Sprintf("%s-%d", cfg.Cluster.ClusterName, time.Now().UnixMicro())
+	}
+
+	cnt := 0
+	if len(cfg.Cluster.BootstrapClusters) != 0 {
+		cnt++
+	}
+	if cfg.Cluster.BootstrapDurl != "" {
+		cnt++
+	}
+	if cfg.Cluster.BootstrapDdns != "" {
+		cnt++
+	}
+	if cnt > 1 {
+		return nil, errors.New("you can only pass one 'bootstrap_xxx' to bootstrap the node, or leave them empty to start a single-node cluster")
+	}
```

Comment:
Why make it so complicated, are other components also like this?

Reply:
This is common for etcd and other cloud native applications. Other pingcap components relies on the central PD(they only need PD).

PD only provide `bootstrapClusters` options, and a sugar option of `bootstrapCluster` by `-join xxx:2380` if you don't want to write all clusters by hands. I have no plan to include that sugar option in this PR.

Honestly, discovery or dns is more convenient.

To use `--initial-clusters`, you need to specify all nodes. To use `-join`, the cluster must be bootstraped already. Whatever you need to pass different options fot different PD and probably follow some orders.

Operator has implemented some very complex code.

Reply:
In my option, there's even no need to merge this PR, right? Proxies don't need to know each other, except for updating configurations. But updating configurations can be achieved by configuring the etcd address.

Reply:
Only if you want to remove etcd from tiproxy, and relies on an external ETCD. I think it makes things complicated. If you have PD, you don't want another etcd. If you use tiproxy with static servers, you need another etcd.